src/app/content/hooks/receivePageNotFoundId.spec.ts

```diff
@@ -0,0 +1,67 @@
+import createTestServices from '../../../test/createTestServices';
+import createTestStore from '../../../test/createTestStore';
+import { MiddlewareAPI, Store } from '../../types';
+import { receivePageNotFoundId } from '../actions';
+
+const mockFetch = (valueToReturn: any, error?: any) => () => new Promise((resolve, reject) => {
+  if (error) {
+    reject(error);
+  }
+  resolve({ json: () => new Promise((res) => res(valueToReturn)) });
+});
+
+describe('receivePageNotFoundId hook', () => {
+  let hook: ReturnType<typeof import ('./receivePageNotFoundId').receivePageNotFoundIdHookBody>;
+  let store: Store;
+  let helpers: MiddlewareAPI & ReturnType<typeof createTestServices>;
+  let historyReplaceSpy: jest.SpyInstance;
+  let fetchBackup: any;
+
+  beforeEach(() => {
+    store = createTestStore();
+
+    helpers = {
+      ...createTestServices(),
+      dispatch: store.dispatch,
+      getState: store.getState,
+    };
+
+    helpers.history.location = {
+      pathname: '/books/physics/pages/1-introduction301',
+    } as any;
+
+    historyReplaceSpy = jest.spyOn(helpers.history, 'replace');
```

Comment:
try throwing a `mockImplementation` on this, i wonder if the history implementation is causing issues for you

Reply:
it didn't help, i'll keep trying